src/ADAL.PCL/Flows/AcquireTokenInteractiveHandler.cs

```diff
@@ -79,9 +81,24 @@ public AcquireTokenInteractiveHandler(RequestData requestData, Uri redirectUri,
             this.UniqueId = userId.UniqueId;
             this.DisplayableId = userId.DisplayableId;
             this.UserIdentifierType = userId.Type;
-            this.LoadFromCache = (requestData.TokenCache != null && parameters != null && PlatformPlugin.PlatformInformation.GetCacheLoadPolicy(parameters));
             this.SupportADFS = true;
 
+            claims = ProcessClaims(extraQueryParameters, claims);
+            if (!String.IsNullOrEmpty(claims))
+            {
+                PlatformPlugin.Logger.Verbose(CallState,
+                string.Format(CultureInfo.InvariantCulture,
+                    "Claims present. Skip cache lookup."));
+            }
+            this.LoadFromCache = (requestData.TokenCache != null && parameters != null && PlatformPlugin.PlatformInformation.GetCacheLoadPolicy(parameters) && String.IsNullOrEmpty(claims));
+
+            // Push claims into extraQueryParameters.
+            if (!String.IsNullOrEmpty(this.extraQueryParameters))
+            {
+                this.extraQueryParameters += "&";
+            }
+            this.extraQueryParameters += "claims=" + claims;
```

Comment:
I don't think this is necessary. We're not sending claims as eqp. I'm removing this part.